src/wp-includes/class-wp-roles.php

```diff
@@ -160,6 +160,13 @@ public function add_role( $role, $display_name, $capabilities = array() ) {
 			return;
 		}
 
+		foreach ( $capabilities as $key => $value ) {
+			if ( ! is_bool( $value ) ) {
+				$capabilities[ $value ] = true;
+				unset( $capabilities[ $key ] );
+			}
+		}
```

Comment:
Rather than checking every value individually, I think it would make more sense to check the entire `$capabilities` array for the shape that is provides. While there's arguably a reason to _either_ specify a `$cap => $grant` map _or_ a list of caps, it makes no sense to me to allow a mix of the two - that would just be unintuitive.

So I think we could simplify this by using `wp_is_numeric_array()`. For example:
```suggestion
		if ( wp_is_numeric_array( $capabilities ) ) {
			$capabilities = array_fill_keys( $capabilities, true );
		}
```

Reply:
If it were up to me, we would deprecate supporting a map entirely, as the absence of a capability implies it is not granted. Without explicitly deprecating that usage, I think it makes sense to support both, even in a single call. Is there a performance implication here? I won't disagree that it's kind of gross.

Reply:
After discussion with @johnbillion we're going to move forward w/o support for a mixed format array